src/helpers/ApiServiceHelper.js

```diff
@@ -29,7 +29,143 @@ function ApiServiceHelper (url = '', clientId = '', clientSecret = '', scope = '
+          type: 'empty-object-property-data',
+          function: functionName,
+          error: { nyplSource: key, data: arrayOfDecodedRecords }
+        });
+      }
+
+      // Iterate over each record data to extract the record id
+      Object.keys(arrayOfDecodedRecords).forEach((k) => {
+        // Verify the existence of the 'record' property
+        if (!arrayOfDecodedRecords[k].hasOwnProperty('record') || !arrayOfDecodedRecords[k].record) {
+          throw HoldRequestConsumerError({
+            message: 'the record object key within the decoded data is not defined',
+            type: 'empty-object-property-record',
+            function: functionName,
+            error: { nyplSource: key, data: arrayOfDecodedRecords[k] }
+          });
+        }
+
+        // Append the record value (id) to url string
+        fullItemUrl += arrayOfDecodedRecords[k].record;
+
+        // Append a comma (,) to all except the last item
+        if (arrayOfDecodedRecords.length !== (parseInt(k) + 1)) {
+          fullItemUrl += ',';
+        }
+      });
+
+      // Append the nyplSource to the end of the string
+      fullItemUrl += '&nyplSource=' + key;
+
+      if (records[key].hasOwnProperty('item-api') && records[key]['item-api']) {
+        throw HoldRequestConsumerError({
+          message: 'the item-api object key within decodedData already exists',
+          type: 'existent-object-property-item-api',
+          function: functionName,
+          error: { nyplSource: key, data: records[key]['item-api'] }
+        });
+      }
+
+      // Extend the source object to include the itemApiUrl
+      records[key]['item-api'] = fullItemUrl;
+
+      // Reset the item URL for the next record
+      fullItemUrl = itemApiUrl;
+    });
+
+    return records;
+  };
+
+  this.getItemData = (records, apiUrl) => {
+    // Scaffold for API Call to Item service
+  };
+
+  this.groupRecordsBy = (records, key) => {
+    const functionName = 'groupRecordsBy';
+
+    if (!records) {
+      throw HoldRequestConsumerError({
+        message: 'the records array is not defined',
+        type: 'undefined-records-array-parameter',
+        function: functionName
+      });
+    }
+
+    if (!records.length) {
+      throw HoldRequestConsumerError({
+        message: 'the records array is empty',
+        type: 'empty-records-array-parameter',
+        function: functionName
+      });
+    }
+
+    if (!key || key === '') {
+      throw HoldRequestConsumerError({
+        message: 'the key parameter is not defined or empty',
+        type: 'undefined-key-parameter',
+        function: functionName
+      });
+    }
+
+    const matchingRecordsObject = records.reduce((allRecords, record) => {
+      allRecords[record[key]] = allRecords[record[key]] || { 'data': [] };
+      allRecords[record[key]]['data'].push(record);
+      return allRecords;
+    }, {});
+
+    return matchingRecordsObject;
+  };
+
   this.getOAuthToken = (cachedToken) => {
```

Comment:
Would it be good to break out the OAuth functionality into another function? More in line with the *S* of SOLID (https://scotch.io/bar-talk/s-o-l-i-d-the-first-five-principles-of-object-oriented-design)?